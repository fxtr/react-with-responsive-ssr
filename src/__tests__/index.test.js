import React from 'react';
import { render } from '@testing-library/react';
import withResponsive, { setMediaQueryBreakpoints } from '../index';

// @todo
// mock `styled-jsx/css` `resolve()` to have the resulting css in snapshot
// tried to mock but it didn't worked only for this
// try again later

const TestComponent = ({ className, ...props }) => (
  <div className={`test-component ${className}`} {...props} />
);

describe('renders with default breakpoints', () => {
  it('with props', () => {
    const ResponsiveComponent = withResponsive(TestComponent);
    const comps = {
      'from-sm': render(<ResponsiveComponent displayFrom="sm">From sm</ResponsiveComponent>),
      'to-lg': render(<ResponsiveComponent displayTo="lg">To lg</ResponsiveComponent>),
      'from-md-to-xl': render(
        <ResponsiveComponent displayFrom="md" displayTo="xl">
          From md - To xl
        </ResponsiveComponent>
      ),
    };
    Object.keys(comps).forEach(k => {
      const {
        container: { firstChild },
      } = comps[k];
      expect(firstChild).toMatchSnapshot(`default-breakpoints:${k}`);
    });
  });

  it('with inline props', () => {
    const FromSM = withResponsive(TestComponent, { displayFrom: 'sm' });
    const ToLG = withResponsive(TestComponent, { displayTo: 'lg' });
    const FromMDToXL = withResponsive(TestComponent, { displayFrom: 'md', displayTo: 'xl' });
    const comps = {
      'from-sm': render(<FromSM>From sm</FromSM>),
      'to-lg': render(<ToLG>To lg</ToLG>),
      'from-md-to-xl': render(<FromMDToXL>From md - To xl</FromMDToXL>),
    };
    Object.keys(comps).forEach(k => {
      const {
        container: { firstChild },
      } = comps[k];
      expect(firstChild).toMatchSnapshot(`default-breakpoints:${k}`);
    });
  });
});

describe('renders with custom breakpoints', () => {
  beforeAll(() =>
    setMediaQueryBreakpoints({
      mobile: 0,
      tablet: 1024,
      desktop: 1440,
    })
  );

  it('with props', () => {
    const ResponsiveComponent = withResponsive(TestComponent);
    const comps = {
      'from-mobile': render(<ResponsiveComponent displayFrom="mobile">From mobile</ResponsiveComponent>),
      'to-desktop': render(<ResponsiveComponent displayTo="desktop">To desktop</ResponsiveComponent>),
      'from-tablet-to-desktop': render(
        <ResponsiveComponent displayFrom="tablet" displayTo="desktop">
          From tablet - To desktop
        </ResponsiveComponent>
      ),
    };
    Object.keys(comps).forEach(k => {
      const {
        container: { firstChild },
      } = comps[k];
      expect(firstChild).toMatchSnapshot(`custom-breakpoints:${k}`);
    });
  });

  it('with inline props', () => {
    const FromMobile = withResponsive(TestComponent, { displayFrom: 'mobile' });
    const ToDesktop = withResponsive(TestComponent, { displayTo: 'desktop' });
    const FromTabletToDesktop = withResponsive(TestComponent, {
      displayFrom: 'tablet',
      displayTo: 'desktop',
    });
    const comps = {
      'from-mobile': render(<FromMobile>From mobile</FromMobile>),
      'to-desktop': render(<ToDesktop>To desktop</ToDesktop>),
      'from-tablet-to-desktop': render(<FromTabletToDesktop>From tablet - To desktop</FromTabletToDesktop>),
    };
    Object.keys(comps).forEach(k => {
      const {
        container: { firstChild },
      } = comps[k];
      expect(firstChild).toMatchSnapshot(`default-breakpoints:${k}`);
    });
  });
});
