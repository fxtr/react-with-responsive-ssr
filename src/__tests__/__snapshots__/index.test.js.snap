// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom breakpoints with inline props: default-breakpoints:from-mobile 1`] = `
<div
  class="test-component "
>
  From mobile
</div>
`;

exports[`renders with custom breakpoints with inline props: default-breakpoints:from-tablet-to-desktop 1`] = `
<div
  class="test-component jsx-3298173018"
>
  From tablet - To desktop
</div>
`;

exports[`renders with custom breakpoints with inline props: default-breakpoints:to-desktop 1`] = `
<div
  class="test-component jsx-3747896157"
>
  To desktop
</div>
`;

exports[`renders with custom breakpoints with props: custom-breakpoints:from-mobile 1`] = `
<div
  class="test-component "
>
  From mobile
</div>
`;

exports[`renders with custom breakpoints with props: custom-breakpoints:from-tablet-to-desktop 1`] = `
<div
  class="test-component jsx-3298173018"
>
  From tablet - To desktop
</div>
`;

exports[`renders with custom breakpoints with props: custom-breakpoints:to-desktop 1`] = `
<div
  class="test-component jsx-3747896157"
>
  To desktop
</div>
`;

exports[`renders with default breakpoints with inline props: default-breakpoints:from-md-to-xl 1`] = `
<div
  class="test-component jsx-1213670158"
>
  From md - To xl
</div>
`;

exports[`renders with default breakpoints with inline props: default-breakpoints:from-sm 1`] = `
<div
  class="test-component jsx-2917367701"
>
  From sm
</div>
`;

exports[`renders with default breakpoints with inline props: default-breakpoints:to-lg 1`] = `
<div
  class="test-component jsx-242699950"
>
  To lg
</div>
`;

exports[`renders with default breakpoints with props: default-breakpoints:from-md-to-xl 1`] = `
<div
  class="test-component jsx-1213670158"
>
  From md - To xl
</div>
`;

exports[`renders with default breakpoints with props: default-breakpoints:from-sm 1`] = `
<div
  class="test-component jsx-2917367701"
>
  From sm
</div>
`;

exports[`renders with default breakpoints with props: default-breakpoints:to-lg 1`] = `
<div
  class="test-component jsx-242699950"
>
  To lg
</div>
`;
